js/js_test.go

```diff
@@ -571,3 +572,207 @@ func TestUint8Array(t *testing.T) {
 		t.Errorf("Non-empty byte array is not externalized as a Uint8Array")
 	}
 }
+
+// Internalize is used as a helper type to test $internalize. A struct is used
+// in order that all of the Go types can be verified (otherwise we are limited
+// to the methods on *Object). Where methods on *Object exist they too will be
+// tested
+type Internalize struct {
+	*js.Object
+
+	string string `js:"string"`
+	bool   bool   `js:"bool"`
+}
+
+func TestInternalizeString(t *testing.T) {
+	fieldName := "string"
+	zero := ""
+
+	s := &Internalize{Object: js.Global.Get("Object").New()}
```

Comment:
Stylistic note:  I wonder if a table-based test would make sense here?  It may make the tests more readable, reduce the need for the `// ***` comment blocks, and if coupled with `t.Run(...)` calls for each test, could make the output more readable.

Reply:
@flimzy just pushed up a code-generated version of the tests which seems to work well. Let me know what you think.